crates/weaver_resolver/src/attribute.rs

```diff
@@ -23,12 +24,55 @@ pub struct AttributeCatalog {
 }
 
 #[derive(Debug, PartialEq)]
-struct AttributeWithGroupId {
+/// The Attribute with its group ID.
+pub struct AttributeWithGroupId {
+    /// The attribute.
     pub attribute: attribute::Attribute,
+    /// The group ID.
     pub group_id: String,
 }
 
+/// A reference to an attribute with its original attribute and group ID.
+pub struct AttributeRefWithGroup {
```

Comment:
Do we really need this if a body field is, in fact, an attribute once resolved? My feeling is that we are adding too much complexity here. Probably the best course of action for now is to finalize the discussion on the aliased attributes and then see if we really need all this code in the AttributeCatalog.

Reply:
Body fields will NEVER be present in the Attribute Catalog as they are NOT attributes.

Different event definitions MAY define the same field "name" but they MAY have semantically different meanings (ie. a Body Field named XXX is explicitly NOT directly comparable with any other Body Field named XXXX UNLESS the event.name is the same.

See https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/events.md#general-event-semantics and explicitly the section `Recommendations on using attributes vs. body fields`

Reply:
Removing as the no longer needed with the removal of `ref` / `alias` option